src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/yflows/YFlowUpdateSpec.groovy

```diff
@@ -108,6 +111,92 @@ class YFlowUpdateSpec extends HealthCheckSpecification {
         ]
     }
 
+    @Tidy
+    def "User can update y-flow where one of subflows has both ends on shared switch"() {
+        given: "Existing y-flow where one of subflows has both ends on shared switch"
+        def switchTriplet = topologyHelper.getSwitchTriplets().first()
+        switchTriplet.setEp1(switchTriplet.getShared())
```

Comment:
This is not the best way to get a switch triplet you require. Right now you have a 'triplet' object with messed up paths and endpoints (you manually changed the ep1, but path objects remain for the previous ep1). So basically we may want to enforce triplet to be Immutable to now allow such code constructions (remove setters?). What you want to do, is to _find_ the desired triplet:
```
def switchTriplet = topologyHelper.getSwitchTriplets().find { it.ep1 == it.shared }
```
This way your object will be complete, with 'paths' matching the endpoints etc.
This does not impact this particular testcase (not 100% sure), but it's not a good idea to have this illegal construction anywhere in the codebase, as it can backfire in any other subsequent helper method call that may want to access 'paths' for this obj

Reply:
Got the picture. Updated. Please, review.